stan/math/rev/mat/fun/cholesky_decompose.hpp

```diff
@@ -241,9 +384,19 @@ inline Eigen::Matrix<var, -1, -1> cholesky_decompose(
   check_symmetric("cholesky_decompose", "A", A);
 
   Eigen::Matrix<double, -1, -1> L_A(value_of_rec(A));
+#ifdef STAN_OPENCL
+  if (L_A.rows() > opencl_context.tuning_opts().cholesky_size_worth_transfer) {
+    // NOTE: We don't do the pos def check here because
+    // The GPU cholesky only returns the lower left
+    L_A = cholesky_decompose(L_A);
```

Comment:
Just to clarify this, we do check if the input was pos def in the same way that Eigen does [here](https://github.com/stan-dev/math/blob/develop/stan/math/prim/mat/err/check_pos_definite.hpp#L84), its just that we do that inside the cholesky_decompose call. 

check_pos_definite checks if the decomposition was successful (no numerical issues, basically no NaNs) and if the diagonal is non-zero which is the same as we do [here](https://github.com/stan-dev/math/blob/develop/stan/math/gpu/cholesky_decompose.hpp#L92).

Reply:
Thanks Rok! I'll update the comment